src/Evolution/Systems/RelativisticEuler/Valencia/PrimitiveFromConservative.cpp

```diff
@@ -0,0 +1,181 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/RelativisticEuler/Valencia/PrimitiveFromConservative.hpp"
+
+#include <cmath>
+#include <limits>
+#include <type_traits>
+
+#include "DataStructures/DataVector.hpp"  // IWYU pragma: keep
+#include "DataStructures/Tensor/EagerMath/DotProduct.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "NumericalAlgorithms/RootFinding/TOMS748.hpp"
+#include "PointwiseFunctions/GeneralRelativity/IndexManipulation.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/Overloader.hpp"
+
+// IWYU pragma: no_forward_declare EquationsOfState::EquationOfState
+
+/// \cond
+namespace RelativisticEuler {
+namespace Valencia {
+
+namespace {
+
+template <typename DataType, size_t ThermodynamicDim>
+class FunctionOfZ {
+ public:
+  FunctionOfZ(const Scalar<DataType>& tilde_d,
+              const Scalar<DataType>& tilde_tau,
+              const Scalar<DataType>& tilde_s_magnitude,
+              const std::unique_ptr<
+                  EquationsOfState::EquationOfState<true, ThermodynamicDim>>&
+                  equation_of_state)
+      : tilde_d_(tilde_d),
+        tilde_tau_(tilde_tau),
+        tilde_s_magnitude_(tilde_s_magnitude),
+        equation_of_state_(equation_of_state) {}
+  double operator()(const double z, const size_t i = 0) const noexcept {
+    const double r =
```

Comment:
I see that these variable names match the ones in the documentation, but if only to be consistent with our guidelines and to make it easier to search for these variables, should we think of using a name like `variable_r` or similar in cases like these? (If there is no better long name that can be thought of.)